frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx

```diff
@@ -105,7 +105,6 @@ export function PropertyValue({
                             values: [...Array.from(new Set(propValues))],
                             status: 'loaded',
                         },
-                        ...options,
```

Comment:
I'm not sure I understand how removing this fixes things. Does `propertyKey` even change in the demos? We'll delete the previously cached results for other property keys, but those are not even displayed here, as you're looking at just one property...?

What I guess is happening is out-of-order events, the stuff that kea's `breakpoint`s are built to resolve. Here you should figure out if "newInput" when requested is "newInput" as it's now... and only set the options if the returned response if for the last made response.

Reply:
Yep, I guess would be a pain to write tests without converting to a logic (or at least should convert to a logic if making that much effort)

Here what is being changed are the items being returned from a search for combination of property key and property value

For clarity here `options` is the value already held before the search and the object before it (lines 105 and 106) are the results of the search

So 

 * we get the results of a search (the backend has done the work of filtering them), 
 * spread them into an object
 * then spread the existing data (which was not returned by the current search) on top of them
 * then [here on line 134](https://github.com/PostHog/posthog/blob/master/frontend/src/lib/components/PropertyFilters/components/PropertyValue.tsx#L134) we filter them in the front end

That spread and filter means that we don't display the results of the search to the user. When the logical thing is that the user sees the API result. If they want the previous results they can clear the input

There are several confusing things about the behaviour on this box. This is only one of them

Reply:
Isn't the problem that:

```ts
setOptions({
    [propertyKey]: { // <--- this `[propertyKey]` here
        values: [...Array.from(new Set(outerOptions))],
        status: 'loaded',
    },
    ...options, // <-- gets overridden by the previous value of the same key here
})
```

If so, then 

1. Moving `...options,` before the `[propertyKey]: {},` is the true fix. Not just here, but in the 2 other places that call this.
2. Removing `...options` is a fine solution as well, but then this should be explicitly phrased as a PR to remove "other key" caching from the property value. In addition I'd consider de-nesting "options" from `{ onlyOneKey: { value } }` to `value`, and also update the other `setOptions` that suffer from this.